src/components/invoices/InvoiceCreditNotesTable.tsx

```diff
@@ -0,0 +1,342 @@
+import React, { memo } from 'react'
+import { gql } from '@apollo/client'
+import styled from 'styled-components'
+import { generatePath } from 'react-router-dom'
+
+import { Typography } from '~/components/designSystem'
+import { CreditNote, CreditNoteItem, CurrencyEnum } from '~/generated/graphql'
+import { useInternationalization } from '~/hooks/core/useInternationalization'
+import { HEADER_TABLE_HEIGHT, NAV_HEIGHT, theme } from '~/styles'
+import { intlFormatNumber } from '~/core/formats/intlFormatNumber'
+import { CUSTOMER_INVOICE_CREDIT_NOTE_DETAILS_ROUTE } from '~/core/router'
+
+gql`
+  fragment InvoiceForCreditNotesTable on Invoice {
+    id
+    totalAmountCents
+    customer {
+      id
+    }
+    creditNotes {
+      id
+      creditAmountCents
+      creditAmountCurrency
+      number
+      subTotalVatExcludedAmountCents
+      subTotalVatExcludedAmountCurrency
+      vatAmountCents
+      vatAmountCurrency
+      items {
+        amountCents
+        fee {
+          id
+          amountCents
+          amountCurrency
+          eventsCount
+          units
+          feeType
+          charge {
+            id
+            billableMetric {
+              id
+              name
+              aggregationType
+            }
+          }
+          subscription {
+            id
+            name
+            plan {
+              id
+              name
+            }
+          }
+          group {
+            id
+            key
+            value
+          }
+        }
+      }
+    }
+  }
+`
+
+interface InvoiceCreditNotesTableProps {
+  customerId: string
+  invoiceId: string
+  formatedCreditNotes: {
+    creditNote: CreditNote
+    items: CreditNoteItem[]
+  }[]
+  totalAmountCents: number
+}
+
+export const InvoiceCreditNotesTable = memo(
+  ({
+    customerId,
+    formatedCreditNotes,
+    invoiceId,
+    totalAmountCents,
+  }: InvoiceCreditNotesTableProps) => {
+    const { translate } = useInternationalization()
+
+    return (
+      <Wrapper>
+        {formatedCreditNotes.map((formatedCreditNote, i) => {
+          const creditNote = formatedCreditNote.creditNote
+          const substractRemainingAmount = (initialValue: number, valueToRemove: number) => {
+            return initialValue - valueToRemove
+          }
+          let remainingAmountAfterCreditNote = totalAmountCents || 0
+
+          for (let l = 0; l < i + 1; l++) {
+            const element = formatedCreditNotes[l]
+
+            remainingAmountAfterCreditNote = substractRemainingAmount(
+              remainingAmountAfterCreditNote,
+              // @ts-ignore
+              Number(element.creditNote.creditAmountCents)
+            )
+          }
+
+          return formatedCreditNote?.items.map((subscriptionItem, j) => {
+            // @ts-ignore
```

Comment:
Why so many ts-ignore ?

Reply:
wasn't defining any type for this custom implementation. Fixed now